packages/codemods/documentation/howTo.md

```diff
@@ -0,0 +1,153 @@
+- Handling that `Result` return type isn't too tricky! Take a look at the return types of the utilities you're given -- they're often results too, which allows for you to pick out error / success messages and wrap them in result you return. To return a `Result` of type `ModResult, NoOp`, you'll simply have to return the constructor `Ok({ logs: [some success messages]})` when you know a mod has completed running, or `Err({ reason: 'why this mod failed' })`, if you encounter a place where you know a mod has failed. Feel free to checkout existing codemods for examples.
+
+### Creating a CodeMod Manually
+
+- There is also a third way to create a codemod, which is how most of the existing codemods are written. For an example, check out [../src/codemods/mods/oldToNewButton/oldToNewButton.mod.ts](../src/codemods/mods/oldToNewButton/oldToNewButton.mod.ts).
+  - Namely, this method actually explicitly creates a codemod object, allowing for the most flexibility possible. Here's a template for making your own:
+
+```typescript=
+//some imports
+
+const newCodeModName: CodeMod = {
+  run: (file: SourceFile) => {
+    try {
+      // Body of your codemod
+    } catch (e) {
+      return Err({ reason: /* display error e */ });
+    }
+    return Ok({ logs: [/* list of made changes */] });
+  },
+  version: /*some version string*/,
+  name: 'newCodeModName',
+  enabled: true,
+};
+
+export default newCodeModName;
+```
+
+- If you export the mod name as the default object, it will get detected and run if you run all codemods!
+
+### Good Mod-Writing Practices
+
+- Because codemod utilities aren't always perfect (due to bugs / limitations of ts-morph), it's very important to have effective logging so that clients / testers know what worked and what didn't when modding.
+- This primarily involves returning a meaningful codemod result from your codemod `run()` function. Inside `run()`, the better you log your successes and failures (by either compiling the changelog of a given mod in a list and returning it in Ok() or returning an actionable error message in Err()), the easier time devs in the future will have when working with your mod!
+
+### Testing Your CodeMods
+
+###### If you're writing your first codemod, do this work in `configMod.test.ts`.
+
+- Before you run your codemod, you'll need to specify files to run on! The following code will do:
+
+```typescript=
+project = new Project();
+project.addSourceFileAtPaths(`${process.cwd()}addSomePathHere!`);
+/* If you want to add many paths, you can do that too. */
+project.addSourceFilesAtPaths(`${process.cwd()}someRegexPath`);
+```
+
+- Once you have your codemod(s), to run them, you can use the function `runMods()`, which accepts an array of codemods, an array of source files, and a callback function that reports on the success of the mod. Here's what an invocation might look like:
```

Comment:
This all happens automatically, you shouldn't need to worry about how runMods works.

Reply:
That makes sense!